### PR TITLE
Metrics Driven Decisions

### PR DESCRIPTION
**Pull request by Eli1123**
*Issue #, if available:*

*Description of changes:*
Added metrics driven decisions to CUE, added a load testing docker file for the rust application, added the sample rust application, and a doc to describe more on what I did.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: platform/components/appmod-service.cue [hunk 1]
```diff
@@ -190,11 +295,11 @@ template: {
 											"spec": {
 												"containers": [
 													{
-														"name": "test",
+														"name": "artillery-container",
```
Comment (shapirov103) [inline platform/components/appmod-service.cue]: This cannot be artillery container - it is a generic cue template for all applications.

File: deployment/teams/team-r/dev/rust-application.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        replicas: 5
+        serviceAccount: "rust-service-account"
+        dummyTestVariable: "test-#"
+        functionalGate:
+          pause: "10s" 
+          image: "699475918442.dkr.ecr.us-west-2.amazonaws.com/modernengg/rust-test:latest"
+        performanceGate:
+          pause: "5s"
+          image: "699475918442.dkr.ecr.us-west-2.amazonaws.com/modernengg/rust-test:latest"
+        functionalMetric:
```
Comment (shapirov103) [inline deployment/teams/team-r/dev/rust-application.yaml]: let's change `functionalMetric`  to just `metrics` (plural)

File: platform/components/appmod-service.cue [hunk 3]
```diff
@@ -219,15 +324,29 @@ template: {
...
+	}
+
 	parameter: {
         image_name: string
         image: string
         replicas: *3 | int
         port: *80 | int
         targetPort: *8080 | int
 		serviceAccount: *"default" | string
+		dummyTestVariable?: string
```
Comment (shapirov103) [inline platform/components/appmod-service.cue]: why do we need this dummy?

File: platform/components/appmod-service.cue [hunk 4]
```diff
@@ -219,15 +324,29 @@ template: {
...
 		functionalGate?: #QualityGate
 		performanceGate?: #QualityGate
+		functionalMetric?: #MetricGate
```
Comment (shapirov103) [inline platform/components/appmod-service.cue]: functionalMetric -> metrics

**Review by shapirov103 — COMMENTED** [review]
@Eli1123 great work. 

Please my comments to address  - just a few. 

Also in order for your changes to work properly, you need to generate the component definition yaml for the appmod-service.cue and put it under https://github.com/aws-samples/appmod-blueprints/tree/main/deployment/addons/kubevela/templates/components

Otherwise the result of your work won't apply.

File: platform/components/appmod-service.cue (hunk 1 above)
Comment (Eli1123) [reply platform/components/appmod-service.cue]: Ah yes forgot to change that back.

File: platform/components/appmod-service.cue (hunk 3 above)
Comment (Eli1123) [reply platform/components/appmod-service.cue]: Allows for them to trigger a rollout for testing purposes easily so I figured it was worth an addition.

File: deployment/teams/team-r/dev/rust-application.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        componentNamesForAccess:
+          - dynamodb-table
+        clusterName: modernengg-dev
+        clusterRegion: us-west-2
+        dependsOn:
+          - dynamodb-table
+    - name: rust-backend
+      type: appmod-service
+      properties:
+        image:  699475918442.dkr.ecr.us-west-2.amazonaws.com/modernengg/rust-microservice:latest # Target image for workload
```
Comment (shapirov103) [inline deployment/teams/team-r/dev/rust-application.yaml]: remove explicit references to your account. Consider it a template, you can use image as <image>.

Applies elsewhere

File: applications/rust/integration/demo-build/benchmark.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+config:
+  target: "http://127.0.0.1:80" #TODO: Override with artillery command
```
Comment (shapirov103) [inline applications/rust/integration/demo-build/benchmark.yaml]: Do we need this TODO here?

File: applications/rust/integration/demo-build/manual-load-test-app.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: rust-svc-load-test
+  labels:
+    workload: rust-backend
+spec:
+  containers:
+    - name: artillery-container
```
Comment (shapirov103) [inline applications/rust/integration/demo-build/manual-load-test-app.yaml]: explicit reference to account in the image field

File: applications/rust/integration/demo-build/manual-load-test-app.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
```
Comment (shapirov103) [inline applications/rust/integration/demo-build/manual-load-test-app.yaml]: Do we need this pod definition? Seems like an artifact from the testing phase.

**Review by shapirov103 — COMMENTED** [review]
A couple more comments, please take a look

File: applications/rust/integration/demo-build/benchmark.yaml (hunk 6 above)
Comment (Eli1123) [reply applications/rust/integration/demo-build/benchmark.yaml]: Shardul had it on his benchmark file I got this from so not sure. Might just be a left over.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**